### PR TITLE
Add support for GraphicsMagick

### PR DESCRIPTION
**Pull request by cblock**
Attached please find my initial attempt at integrating GraphicsMagick into the lib as proposed in #7.

Please note that there GraphicsMagick handles EXIF data slightly differently. Instead of providing `exif:DateTimeOriginal` keys it provides `DateTimeOriginal` keys. I thus added a little normalization logic that ensures EXIF Data ist always provided in GraphicsMagick format no matter what library is used underneath.

### CODE REVIEW

**Comment by cblock** [comment]
Not quite sure why travis fails. Looks like GM executable cannot be called though graphicsmagick seem to have been installed properly. @keichan34 maybe you have an idea?

File: lib/exfile_imagemagick/sys_runner.ex [hunk 1]
```diff
@@ -1,9 +1,20 @@
 defmodule ExfileImagemagick.SysRunner do
   alias ExfileImagemagick.Server
 
-  def cmd(command, args) do
+  @image_processor Application.get_env(:exfile_imagemagick, :image_processor, :imagemagick)
```
Comment (keichan34) [inline lib/exfile_imagemagick/sys_runner.ex]: If you move this to `ExfileImagemagick.Server`, the code could be cleaned up a bit.

File: lib/exfile_imagemagick/sys_runner.ex [hunk 2]
```diff
@@ -1,9 +1,20 @@
...
+
+  def generic_cmd(command, args) do
```
Comment (keichan34) [inline lib/exfile_imagemagick/sys_runner.ex]: If generic_cmd won't be used, we don't need it

**Review by keichan34 — COMMENTED** [review]
It looks like there are a couple ways to add graphicsmagick to the Travis build -- can you try a couple of them out?

for example, https://github.com/travis-ci/travis-ci/issues/3128

**Comment by cblock** [comment]
@keichan34 Thanks for your code review and the suggestions. I moved the image procesor logic to the GenServer and I also fixed the failing tests in Travis CI.

**Comment by keichan34** [comment]
@cblock Looks good! Thanks!

**Pull request merged**